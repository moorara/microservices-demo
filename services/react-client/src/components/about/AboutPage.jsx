import React from 'react'

const AboutPage = (props) => (
  <section className="hero is-light is-large is-bold">
    <div className="hero-body">
      <div className="container">
        <div className="content">
          <h1 className="title">About</h1>
          <h2 className="subtitle">Control Center Application</h2>
          <p>
            This is a demo application for monitoring hypothetical remote sites in real-time.
            Each site has a number of sensors sending real-time values continuously.
          </p>
          <p>This demo application is built using some <strong>best-practices</strong> and cool technologies.</p>
          <dl>
            <dt>Architecture</dt>
            <dd><strong>Microservices</strong></dd>
            <dd><strong>Docker, Swarm, Kubernetes</strong></dd>
            <dt>Front-End</dt>
            <dd><strong>React, Redux</strong></dd>
            <dd><strong>Bulma</strong></dd>
            <dd><strong>Webpack</strong></dd>
            <dt>Back-End</dt>
            <dd><strong>Go</strong></dd>
            <dd><strong>Node.js, Express.js</strong></dd>
            <dt>Database</dt>
            <dd><strong>MongoDB, ArangoDB</strong></dd>
            <dd><strong>Postgres</strong></dd>
            <dd><strong>RabbitMQ, Redis</strong></dd>
            <dt>API</dt>
            <dd><strong>REST</strong></dd>
            <dt>Gateways</dt>
            <dd><strong>Traefik, Caddy</strong></dd>
            <dt>Monitoring</dt>
            <dd><strong>Fluentd, ElasticSearch, Kibana</strong></dd>
            <dd><strong>Prometheus, Grafana</strong></dd>
          </dl>
        </div>
      </div>
    </div>
  </section>
)

export default AboutPage
