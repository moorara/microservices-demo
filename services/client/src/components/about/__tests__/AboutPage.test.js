import React from 'react'
import { render } from 'enzyme'
import { renderToJson } from 'enzyme-to-json'

import AboutPage from '../AboutPage'

describe('AboutPage', () => {
  test('renders react component', () => {
    const wrapper = render(<AboutPage />)

    expect(renderToJson(wrapper)).toMatchSnapshot()
    expect(wrapper.find('.title').text()).toBe('About')
    expect(wrapper.find('.subtitle').text()).toBe('Control Center Application')
    expect(wrapper.find('.content').text()).toContain('Go')
    expect(wrapper.find('.content').text()).toContain('React')
    expect(wrapper.find('.content').text()).toContain('Redux')
    expect(wrapper.find('.content').text()).toContain('Docker')
    expect(wrapper.find('.content').text()).toContain('Kubernetes')
    expect(wrapper.find('.content').text()).toContain('Prometheus')
    expect(wrapper.find('.content').text()).toContain('ElasticSearch')
    expect(wrapper.find('.content').text()).toContain('Microservices')
  })
})
