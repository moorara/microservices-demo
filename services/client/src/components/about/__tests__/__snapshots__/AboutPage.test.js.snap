// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage renders react component 1`] = `
<section
  class="hero is-light is-large is-bold"
>
  <div
    class="hero-body"
  >
    <div
      class="container"
    >
      <div
        class="content"
      >
        <h1
          class="title"
        >
          About
        </h1>
        <h2
          class="subtitle"
        >
          Control Center Application
        </h2>
        <p>
          This is a demo application for monitoring hypothetical remote sites in real-time. Each site has a number of sensors sending real-time values continuously.
        </p>
        <p>
          This demo application is built using some 
          <strong>
            best-practices
          </strong>
           and cool technologies.
        </p>
        <dl>
          <dt>
            Architecture
          </dt>
          <dd>
            <strong>
              Microservices
            </strong>
          </dd>
          <dd>
            <strong>
              Docker, Swarm, Kubernetes
            </strong>
          </dd>
          <dt>
            Front-End
          </dt>
          <dd>
            <strong>
              React, Redux
            </strong>
          </dd>
          <dd>
            <strong>
              Bulma
            </strong>
          </dd>
          <dd>
            <strong>
              Webpack
            </strong>
          </dd>
          <dt>
            Back-End
          </dt>
          <dd>
            <strong>
              Go
            </strong>
          </dd>
          <dd>
            <strong>
              Node.js, Express.js
            </strong>
          </dd>
          <dt>
            Database
          </dt>
          <dd>
            <strong>
              MongoDB, ArangoDB
            </strong>
          </dd>
          <dd>
            <strong>
              Postgres
            </strong>
          </dd>
          <dd>
            <strong>
              RabbitMQ, Redis
            </strong>
          </dd>
          <dt>
            API
          </dt>
          <dd>
            <strong>
              REST
            </strong>
          </dd>
          <dt>
            Gateways
          </dt>
          <dd>
            <strong>
              Traefik, Caddy
            </strong>
          </dd>
          <dt>
            Monitoring
          </dt>
          <dd>
            <strong>
              Prometheus, Grafana
            </strong>
          </dd>
          <dd>
            <strong>
              Fluentd, ElasticSearch, Kibana
            </strong>
          </dd>
          <dd>
            <strong>
              OpenTracing, Jaeger
            </strong>
          </dd>
        </dl>
      </div>
    </div>
  </div>
</section>
`;
